src/gpt_j/gpt_j_model.rs

```diff
@@ -0,0 +1,808 @@
+// Copyright 2021 The Eleuther AI and HuggingFace Inc. team. All rights reserved.
+// Copyright 2022 Guillaume Becquin
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//     http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::common::activations::Activation;
+use crate::common::dropout::Dropout;
+use crate::common::embeddings::process_ids_embeddings_pair;
+use crate::gpt_j::attention::LayerState;
+use crate::gpt_j::transformer::GptJBlock;
+use crate::pipelines::common::{ModelType, TokenizerOption};
+use crate::pipelines::generation_utils::private_generation_utils::{
+    PreparedInput, PrivateLanguageGenerator,
+};
+use crate::pipelines::generation_utils::{
+    Cache, GenerateConfig, LMHeadModel, LMModelOutput, LanguageGenerator,
+};
+use crate::{Config, RustBertError};
+use rust_tokenizers::tokenizer::Gpt2Tokenizer;
+use rust_tokenizers::vocab::Gpt2Vocab;
+use serde::{Deserialize, Serialize};
+use std::borrow::{Borrow, BorrowMut};
+use tch::nn::{embedding, Linear};
+use tch::{nn, Device, Tensor};
+
+/// # GPT-J Pretrained model weight files
+pub struct GptJModelResources;
+
+/// # GPT-J Pretrained model config files
+pub struct GptJConfigResources;
+
+/// # GPT-J Pretrained model vocab files
+pub struct GptJVocabResources;
+
+/// # GPT-J Pretrained model merges files
+pub struct GptJMergesResources;
+
+impl GptJModelResources {
+    /// Shared under Apache 2.0 license by the EleutherAI contributors at <https://www.eleuther.ai>. Modified with conversion to C-array format.
+    pub const GPT_J_6B: (&'static str, &'static str) = (
+        "gpt-j-6B/model",
+        "https://huggingface.co/EleutherAI/gpt-j-6B/resolve/main/rust_model.ot",
+    );
+    pub const GPT_J_6B_FLOAT16: (&'static str, &'static str) = (
+        "gpt-j-6B/model",
+        "https://huggingface.co/EleutherAI/gpt-j-6B/resolve/float16/rust_model.ot",
+    );
+    pub const GPT_J_TINY_RANDOM: (&'static str, &'static str) = (
```

Comment:
It would be good to clarify the license for this model with the author or the Huggingface team